[TC-10548] SMIT and Seaco feedback (#102)
tradecloud/tradecloud-docs-api-v2: api/json-vs-xml.md

---
description: >-
  Choose between JSON and XML formats in the Tradecloud API. Learn about format support, when to use each, and how to implement both JSON and tXML formats.
---

# JSON versus XML

> The Tradecloud API supports both JSON and XML formats to accommodate different integration requirements and legacy systems. This guide helps you understand the differences and choose the appropriate format for your implementation.

Reviewer: "legacy" is confusing here. Let's make the wording a bit more simple.

```suggestion
The Tradecloud API supports both JSON and XML formats to support different integration requirements. This guide explains how to choose the appropriate format for your implementation.
```

## Overview

The Tradecloud API provides two data formats:

- **JSON**: The default format for all API endpoints
- **tXML**: A proprietary XML format available for specific endpoints

Both formats contain identical data structures and functionality - tXML is a one-to-one mapping of the JSON format.

## When to use each format

### Use JSON when

> - Building new modern integrations (recommended)

Reviewer: This looks a bit weird. It sound like you'd recommend to build modern integrations ;) 
Ofc that's true, but maybe move `recommended` to the "Overview" section

- Implementing REST API clients

> - You need smaller payload sizes

Reviewer: Although this is a pre of JSON compared to XML, I think this is hardly ever a requirement of customers. imo this bullet can be removed.

Author: removed

### Use tXML when

- Integrating with legacy systems that require XML
- Working with enterprise middleware that requires XML
- Your existing infrastructure is XML-based

## JSON Format

JSON is the **default and recommended format** for all Tradecloud API endpoints.

> ### Advantages

Reviewer: I fined this "Advantages" section a bit of AI-bloat. The bullets mentioned are opinionated, and the customer already got the needed info to make a decision in the paragraph above. imo this section can be removed.

Reviewer: I see it's also not consistent:
- no Disadvantages mentioned
- No Advantages / Disadvantages mentioned for tXML

Author: removed

> - Smaller payload size & faster parsing

Reviewer: imo this this not really a concern for our customers.

#### Buyer Endpoints

- [Send order](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/api-connector/specs.yaml#/buyer-endpoints/sendOrderByBuyerRoute)
- [Send single delivery order](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/api-connector/specs.yaml#/buyer-endpoints/sendSingleDeliveryOrderByBuyerRoute)

#### Supplier Endpoints

- [Send order response](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/api-connector/specs.yaml#/supplier-endpoints/sendOrderResponseBySupplierRoute)

#### Webhook Endpoints

- [Order Webhook](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-connector/specs.yaml#/order-webhook%20endpoints/webhookPost)

### Implementation

To use tXML format, set the `Content-Type` header to `application/xml` in your HTTP requests.

### Viewing tXML Examples

You can view tXML examples in the Swagger UI documentation:

1. Navigate to the desired endpoint in the API documentation
2. In the "Parameter content type" dropdown, select "application/xml"
3. The "Example Value" section will show the tXML structure

> #### API Endpoints

Reviewer: This is duplicate of what is listed above.

Author: Removed, also removed the screenshots, as the Swagger UI will change

- [Send order](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/api-connector/specs.yaml#/buyer-endpoints/sendOrderByBuyerRoute)
- [Send single delivery order](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/api-connector/specs.yaml#/buyer-endpoints/sendSingleDeliveryOrderByBuyerRoute)

![Select order API XML content type](../.gitbook/assets/select-order-api-xml-content-type.png)

[Send order response](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/api-connector/specs.yaml#/supplier-endpoints/sendOrderResponseBySupplierRoute)

![Select order response API XML content type](../.gitbook/assets/select-order-response-api-xml-content-type.png)

> 

Author: I rather keep it, so readers can click directly to view the tXML examples.

#### Webhook Endpoints

[Order Webhook](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-connector/specs.yaml#/order-webhook%20endpoints/webhookPost)

![Select order webhook XML content type](../.gitbook/assets/select-order-webhook-xml-content-type.png)

For detailed tXML structure and examples, see the [XML body format documentation](requests.md#xml-body).

## Format Compatibility

Both JSON and tXML formats:

- Contain identical data structures
- Support the same business logic
- Provide the same validation rules
- Return equivalent responses

You can switch between formats without losing functionality, making it easy to migrate from one format to another as your integration requirements evolve.

## Getting Additional tXML Support

If you need tXML support for additional API endpoints not currently listed, please contact [Tradecloud support](../support.md) with your specific requirements.

> ## Next Steps

Reviewer: This is quite nice!
